apps/aehttp/test/aehttp_sc_SUITE.erl

```diff
@@ -4208,3 +4239,259 @@ wait_for_expected_events(Pid, Config, ExpectedEvents) ->
         _ ->
             wait_for_expected_events(Pid, Config, ExpectedEvents1)
     end.
+
+sc_ws_cancel_offchain_update(Cfg0) ->
+    Cfg = sc_ws_open_(Cfg0),
+    #{initiator := #{pub_key := IPubkey},
+      responder := #{pub_key := RPubkey}} = proplists:get_value(participants, Cfg),
+    UpdateOpts =
+        #{ from => aeser_api_encoder:encode(account_pubkey, IPubkey)
+         , to => aeser_api_encoder:encode(account_pubkey, RPubkey)
+         , amount => 1 },
+    InitUpdate =
+        fun(WhoPid) ->
+            ok = ?WS:register_test_for_channel_events(WhoPid, [sign]),
+            ws_send_tagged(WhoPid, <<"channels.update.new">>, UpdateOpts, Cfg),
+            {ok, <<"update">> = Tag, Data} =
+                wait_for_channel_event(WhoPid, sign, Cfg),
+            ok = ?WS:unregister_test_for_channel_events(WhoPid, [sign]),
+            {Tag, Tag, Data}
+        end,
+    cancel_update(InitUpdate, Cfg, can_cancel_ack),
+    sc_ws_close_(Cfg),
+    ok.
+
+sc_ws_cancel_deposit(Cfg0) ->
+    Cfg = sc_ws_open_(Cfg0),
+    InitUpdate =
+        fun(WhoPid) ->
+            ok = ?WS:register_test_for_channel_events(WhoPid, [sign]),
```

Comment:
The function `with_registered_events([sign], [WhoPid], fun() -> ... end)` could be used here.

Reply:
... and in similar patterns below

Reply:
Addressed